pkgs/stdenv/generic/check-meta.nix

```diff
@@ -622,6 +637,49 @@ let
       maintainers =
         attrs.meta.maintainers or [ ] ++ concatMap (team: team.members or [ ]) attrs.meta.teams or [ ];
 
+      identifiers =
+        let
```

Comment:
Something that I'd like to have is a check to make sure people can only set `meta.identifiers.cpeParts` (an assert at this location could work). This ensures people don't accidentally try to set other identifiers without them being properly supported.

Even better: Also make sure people can't set cpeParts attributes that aren't supported.